### PR TITLE
Populate container.SecurityContext.RunAsUser when --user flag is used

### LINKED ISSUE #1926: CLI: `--user` option does not work as expcted
### Bug report

Specifying `--user` flag does not populate the knative service yaml with the supplied value.

### Expected behavior

` kn service create s0 --image knativesamples/helloworld --user 1001` 

should set the value of `RunAsUser` field in container security context to 1001 but it remains unpopulated.

### kn version

```
gvyas-mac:knative gvyas$ kn version
Version:      v1.13.0
Build Date:   2024-01-24 16:48:59
Git Revision: 543522a3
Supported APIs:
* Serving
  - serving.knative.dev/v1 (knative-serving v1.13.0)
* Eventing
  - sources.knative.dev/v1 (knative-eventing v1.13.0)
  - eventing.knative.dev/v1 (knative-eventing v1.13.0)
```

### Knative (serving/eventing) version

> Nightly
> 0.12.x
> 0.11.x
> 0.10.x
> 0.9.x

### PR DESCRIPTION
**Pull request by vyasgun**
## Changes

* Setting container's RunAsUser field based on the `--user` flag at the end so it does not reinitialize the security context

## Reference

Fixes #1926 

**Release Note**

```release-note

```

### CODE REVIEW

File: pkg/kn/flags/podspec.go [hunk 1]
```diff
@@ -418,5 +411,12 @@ func (p *PodSpecFlags) ResolvePodSpec(podSpec *corev1.PodSpec, flags *pflag.Flag
 		}
 	}
 
+	if flags.Changed("user") {
+		err = UpdateUser(podSpec, p.User)
+		if err != nil {
+			return err
+		}
+	}
+
```
Comment (dsimansk) [inline pkg/kn/flags/podspec.go:420]: Is there a reason to change the order of evaluation here?

Comment (dsimansk) [reply pkg/kn/flags/podspec.go:420]: IMO it's very likely SC was override by `UpdateSecurityContext`.

**Comment by dsimansk** [comment]
It would be nice to have a simple unit tests for the change.

**Comment by dsimansk** [comment]
@vyasgun gentle ping. If you could check my comments, pls.

File: pkg/kn/flags/podspec.go (hunk 1 above)
Comment (vyasgun) [reply pkg/kn/flags/podspec.go:420]: The order of evaluation is causing the value of the user flag to get overwritten. I have created an issue for this: https://github.com/knative/client/issues/1926

**Comment by vyasgun** [comment]
Thanks for the review, @dsimansk! Updated the PR with a small change which will test both flags together.

**Comment by dsimansk** [comment]
@vyasgun I still have a small hesitation how to fix the issue actually. Maybe the `--security-context` should behave differently. To be able to merge with existing values etc.

**Pull request merged**